src/goto-harness/recursive_initialization.cpp

```diff
@@ -1008,6 +1008,21 @@ code_blockt recursive_initializationt::build_function_pointer_constructor(
   for(const auto &target : targets)
   {
     auto const assign = code_assignt{dereference_exprt{result}, target};
+    auto sym_to_lookup =
+      target.id() == ID_address_of
+        ?
+        // This is either address of or pointer; in pointer case, we don't
+        // need to do anything
+        to_symbol_expr(to_address_of_expr(target).operands()[0])
+          .get_identifier()
+        : "";
+    // skip referencing globals because the corresponding symbols in the symbol
+    // table are no longer marked as file local.
+    if(has_prefix(id2string(sym_to_lookup), CPROVER_PREFIX "file_local_"))
```

Comment:
:no_entry_sign: Please introduce a global constant for `file_local_` and use it here and in the mangling.